node.py

```diff
@@ -0,0 +1,23 @@
+class Node:
+    def __init__(self, node_dict: dict):
+        self.node_id = node_dict['id']
+        self.parents = node_dict.get('parents')
+
+    def __repr__(self):
+        return self.node_id
+
+    def is_new_node_cyclic(self, node_id_to_node: dict, node_to_check_id: str):
```

Comment:
from abstraction perspective this method not related to Node it should be in Graph